File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isalpha(char c){
        return (c>='a' and c<='z') or (c>='0' and c<='9');
    }
    bool isPalindrome(string s) {
        
        for(auto &x:s){
            if(x>='A' and x<='Z'){
                x = (x-'A')+'a';
            }
        }
        int n = s.length();
        int l = 0, r = n-1;
        while(l<=r){
            while(l < n and not isalpha(s[l]))
                l++;
            while(r >= 0 and not isalpha(s[r]))
                r--;
            if(l <= r and s[l]!=s[r]){
                return false;
            }
            l++;
            r--;
        }
        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        n=len(s)
        s=s.lower()
        l,r= 0,n-1
    #     print(s)
        while l<=r and l>=0 and r<n:
            while l<n and not s[l].isalnum() and l<=r:
                l+=1
            while r>=0 and not s[r].isalnum() and l<=r:
                r-=1
    #         print(l,r)
            if l<n and r>=0 and l<=r and s[l]!=s[r]:
                return False
            l+=1
            r-=1
        return True
        